all_test.go

```diff
@@ -173,112 +189,129 @@ func isPrefix(prefix, str string) bool {
 	return strings.Index(str, prefix) == 0
 }
 
+//func TestMergeLdap(t *testing.T) {
+//	if (*backend_ldap) {
+//		t.Log("Testing ldap merge")
+//		c := setup1(t)
+//		if err := c.Merge(); err != nil {
+//			t.Errorf("An error occurred while trying to merge ldap users with IMQS users: %v", err)
+//		}
+//		teardown(c)
+//	}
+//}
+
 func TestIdentityCaseSensitivity(t *testing.T) {
-	t.Log("Testing case sensitivity")
-	c := setup1(t)
+	if !*backend_ldap {
+		t.Log("Testing case sensitivity")
+		c := setup1(t)
 
-	if _, e := c.CreateUserStoreIdentity("JOE", "JOEusername", "JOEfirstname", "JOElastname", "JOE084", "123"); e == nil || !isPrefix(ErrIdentityExists.Error(), e.Error()) {
-		t.Errorf("CreateIdentity should fail because identities are case-insensitive. Instead, error is %v", e)
-	}
+		if _, e := c.CreateUserStoreIdentity("JOE", "JOEusername", "JOEfirstname", "JOElastname", "JOE084", "123"); e == nil || !isPrefix(ErrIdentityExists.Error(), e.Error()) {
+			t.Errorf("CreateIdentity should fail because identities are case-insensitive. Instead, error is %v", e)
+		}
 
-	teardown(c)
+		Teardown(c)
+	}
 }
 
 func TestRenameIdentity(t *testing.T) {
-	c := setup1(t)
+	if !*backend_ldap {
```

Comment:
Rather just make these functions exit early, if *backend_ldap is true.
Same applies to TestResetPassword, and I'm guessing in other places too.